loader/loader.c

```diff
@@ -492,6 +492,12 @@ static bool loaderAddJsonEntry(const struct loader_instance *inst,
                                LPSTR json_path,    // JSON string to add to the list reg_data
                                DWORD json_size,    // size in bytes of json_path
                                VkResult *result) {
+
+    // Check for and ignore duplicates.
+    if (*reg_data && strstr(*reg_data, json_path)) {
+        return true;
```

Comment:
I agree with notion of de-duplicating.

The code comment above this function says `returns true if filename was appended to reg_data list`. You are returning true without appending.

Either true means "found" and comment should be updated, or you should return false here.

Reply:
return true means success. In this case it's success since the entry is in the list.

Reply:
You are de-duplicating the path to json files. Because of the way that the loader-ICD interface operates, it seems like we should de-duplicate the path to ICDs. I thought that ICD-path de-duplication was already performed by the loader. Not sure if it was lost, is broken, or I'm hallucinating it.

Additionally de-duplicating the path to json files seems like a good idea.

Reply:
OK. Then please update the comment to reflect this reality.